distributed/preloading.py

```diff
@@ -0,0 +1,87 @@
+import atexit
+import shutil
+import logging
+import os
+import sys
+from importlib import import_module
+
+from .compatibility import reload, invalidate_caches, cache_from_source
+
+
+logger = logging.getLogger(__name__)
+
+
+def load_file(path):
+    """ Loads modules for a file (.py, .pyc, .zip, .egg) """
+    directory, filename = os.path.split(path)
+    name, ext = os.path.splitext(filename)
+    names_to_import = []
+    if ext in ('.py', '.pyc'):
+        if directory not in sys.path:
+            sys.path.insert(0, directory)
```

Comment:
I agree that we should handle the path.  I'm concerned that we are also including everything else in that directory in the path when we may not want to.  One way to handle the path without including all other contents of the directory would be to copy the file to the server's `local_directory` attribute.  Although now that I see @pitrou 's solution that might be even simpler.